android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFcmMessagingService.java

```diff
@@ -31,7 +38,40 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
       return;
     }
 
+    ExponentDB.experienceIdToExperience(remoteMessage.getData().get("experienceId"), new ExponentDB.ExperienceResultListener() {
```

Comment:
Does searching for it asynchronously work ok even when the app is terminated? What I'd be afraid of is starting new threads in a time-limited call-to-service. I believe Android catches it and doesn't terminate the app while the asynchronous callback is being executed, but who knows.

Reply:
Just to be on the safe side — do you think this work ok in standalone apps too? Should we add a note somewhere to ensure test this there?

Reply:
So, I've tested it and it seems to work even if the app was killed. But maybe, we should switch to sync the call 🤔

I believe that will work in standalone apps. However, we should test it :D